Clean up _IotMqtt_Deserialize* proof harnesses

feliperodri opened the pull request:
*Description of changes:*

- Updates the name of the proof harnesses to match the function under verification;
- Update all `cbmc-batch.yaml` files;
- Adds missing `cbmc-viewer.json` files;

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

markrtuttle commented:
I recommend closing this pull request.  I don't think this renaming is work the effort (and I personally prefer not having directories and files begin with an underscore).

feliperodri commented:
> I recommend closing this pull request. I don't think this renaming is worth the effort to get the proofs working again (and I personally prefer not having directories and files begin with an underscore).

@markrtuttle I've updated the folder names and removed the underscores. This PR also adds missing `cbmc-viewer.json` files to the deserialized proofs as well as updates the `cbmc-batch.yaml` files, so I recommend thet we keep it.

danielsn commented on cbmc/proofs/IotMqtt_DeserializeConnack/cbmc-viewer.json:
```diff
@@ -0,0 +1,22 @@
+    "IotTaskPool_strerror",
+    "IotTaskPool_TryCancel"
+  ],
+  "proof-name": "_IotMqtt_DeserializeConnack",
```
Why is there an _?

feliperodri replied:
To match exactly the name of the function under verification.
